jukepoot/puppet_parser.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/python
+"""A parser of /var/log/puppet.log.
+
+
+"""
+
+
+import argparse 
+from datetime import datetime
+import dateutil.parser
+import re
+import collections
+import os
+
+
+def main():
+	parser = argparse.ArgumentParser()
+	parser.add_argument(
+		help="parse the given filename", type=str, 
+		dest="filename", action="store")
+	parser.add_argument(
+		"-y","--year", help="parse file by setting start year", 
+		type=int, dest="year", action="store")
+	args = parser.parse_args()
+	if args.year:
+		print("year to use: {0:d}".format(args.year))
+		year = args.year
+
+	else: #build year for date parsing
+		year = datetime.utcfromtimestamp(
+			os.stat(args.filename).st_birthtime).year
+		# print("creation of file: {0:d}".format(year))
+
+	puppet_parser(args.filename,year)
+
+
+def puppet_parser(filepath, year):
+	"""Accepts the puppet file path and parses it out by calling other functions.
+
```

Comment:
Missing docstrings...

```
"""One line describing things.

Args:
  filepath: describe me
  year: describe me
"""
```

Add a Returns section if it returns anything, and even better would be to have:
```
"""One line describing things.

Args:
  filepath (str): describe me
  year (int): describe me
"""
```
This would help a bit more with how you use this function for other users